cloud/spawn.go

```diff
@@ -96,6 +98,18 @@ func (so *SpawnOptions) validate(settings *evergreen.Settings) error {
 	return nil
 }
 
+// ValidateSSHKey errors if the given key does not start with one of the allowed prefixes.
+func ValidateSSHKey(key string) error {
+	validKeys := []string{rsa, dss, ed25519, ecdsa}
+	for _, prefix := range validKeys {
+		if strings.HasPrefix(key, prefix) {
+			return nil
+		}
+	}
+	return errors.Errorf("either an invalid Evergreen-managed key name has been provided,"+
+		"or the key value does not start with one of these options: '%s'", validKeys)
```

Comment:
Suggestion: `or the key value is not one of the valid types: %s`?

For slices, we don't usually need the single quotes around the string since empty/nil slices are formatted as `[]` with `%s`, so the concern about potentially empty strings doesn't apply here.

Reply:
done